bin/pp.ml

```diff
@@ -17,42 +17,49 @@
 let src = Logs.Src.create "cram.pp"
 module Log = (val Logs.src_log src : Logs.LOG)
 
-let run () file section =
-  let t = Mdx.parse_file Normal file in
-  let t = match section with
-    | None   -> t
-    | Some s ->
-      let re = Re.Perl.compile_pat s in
-      match Mdx.filter_section re t with
-      | None   -> []
-      | Some t -> t
+let filter_by_section section lines =
+  match section with
+  | None -> lines
+  | Some s ->
+    let re = Re.Perl.compile_pat s in
+    match Mdx.filter_section re lines with
+    | None -> []
+    | Some lines -> lines
+
+let run () files section =
+  let t =
+    List.map (fun p ->
+      (p, Mdx.parse_file Normal p |> (filter_by_section section))
+    ) files
   in
   match t with
   | [] -> 1
   | _  ->
-    List.iter (function
-        | Mdx.Section _ | Text _ -> ()
-        | Block b ->
-          let b = Mdx.Block.eval b in
-          if not (Mdx.Block.skip b) then (
-            Log.debug (fun l -> l "pp: %a" Mdx.Block.dump b);
-            let pp_lines = Fmt.(list ~sep:(unit "\n") string) in
-            let contents = Mdx.Block.executable_contents b in
-            match b.value with
-            | Toplevel _ -> Fmt.pr "%a\n" pp_lines contents
-            | OCaml      ->
-              Fmt.pr "%a\n%a\n"
-                Mdx.Block.pp_line_directive (file, b.line)
-                pp_lines contents
-            | _          -> ()
-          )
-      ) t;
+    t |> List.iter (fun (file, blocks) ->
```

Comment:
I don't necessarily mind the change of style but could you please leave this (and the other similar change in `bin/test/main.ml`) out of this PR to ease the review?

Reply:
I've split the PR into two commits:
 - https://github.com/realworldocaml/mdx/pull/173/commits/607342e02fe59b91b32703f9fd437b78364a0d46, which contains the new feature;
 - https://github.com/realworldocaml/mdx/pull/173/commits/ca595dc7e5bd40cec4756c1ad822518c2699fe6c, which changes the style. I've taken the opportunity to be more suggestive with the style change; I find this far easier to understand, particularly in the case of nested `List.iter`s.

Happy to split into another PR altogether, but perhaps a commit split is enough?

Reply:
A separate commit is fine! Thanks!